aws-educate-tw/aws-educate-tpet-backend
# Scrum 162 backend post webhooks webhook id trigger send email

## Description

完成trigger-webhook這隻api的邏輯，讓surveycake可以透過這隻api自動寄送信件

## Type of Change

- [ ] 🐛 Bug Fix（修復 bug）
- [ ] 💄 UI/UX（介面或使用者體驗優化）
- [ ] 🔨 Refactor（程式碼重構）
- [ ] 📝 Documentation（文件更新）
- [ ] 🔧 Configuration（配置變更）
- [ ] 🚀 Performance（效能優化）
- [ ] ✅ Test（測試相關）
- [ ] 🔒 Security（安全性相關）

## Related Issues

[SCRUM-162](https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-162)

## Testing

### How to test
1. Make sure you are in the Postman `Harry-local-dev` environment
2. Call the `POST /auth/login` API ([Postman link](https://aws-educate-tw.postman.co/workspace/TPET-API-Development-Workspace~b6915213-64a3-408d-b297-45f472fa0a6b/request/36968695-ac08ee18-f884-4f11-979b-f3dccaaa1bf0?action=share&source=copy-link&creator=36968695&ctx=documentation)) to get a valid JWT token in the **local-dev environment**
3. Call the `POST /webhook API` in the Postman `Harry-local-dev` environment with a valid payload ([Postman API link](https://aws-educate-tw.postman.co/workspace/TPET-API-Development-Workspace~b6915213-64a3-408d-b297-45f472fa0a6b/request/31171060-bbd5ff55-102c-43c0-b565-aedc13d529e5?action=share&creator=31171060&ctx=documentation&active-environment=31171060-a68faedf-68d1-4502-9ef2-3dd830e580eb))
4. You should receive a `webhook_url` from the response, copt that.
5. Set the webhook url in this [surveycake backstage](https://www.surveycake.com/admin/tw/survey/Nq06L/notification), this details of this surveycake matches the payload of `POST /webhook API`  you have just called.
![image](https://github.com/user-attachments/assets/9e4aa65d-469b-43ad-85c8-3b5cf1f47e7e)
6. Fill this [surveycake](https://www.surveycake.com/s/Nq06L) to test whether it works or not.

## Screenshots/Videos
If the settings are correct, you will get a email like this after filling out the [surveycake](https://www.surveycake.com/s/Nq06L).
![image](https://github.com/user-attachments/assets/04e5a0c3-fbf2-472d-a811-e79820eceb88)

## Checklist

- [x] 我已經測試過這些變更
- [ ] 我已經更新相關文件
- [x] 我的程式碼遵循專案的程式碼規範
- [x] 我已經處理所有的 console warnings/errors
- [x] 我已經移除所有的 console.log
- [ ] 我已經新增必要的測試案例
- [x] 所有現有的測試都能通過

## Additional Notes

N/A

[SCRUM-162]: https://aws-educate-cloud-ambassador-dev-team.atlassian.net/browse/SCRUM-162?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
